includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -0,0 +1,1233 @@
+<?php
+/**
+ * Stripe Payment Request API
+ * Adds support for Apple Pay and Chrome Payment Request API buttons.
+ * Utilizes the Stripe Payment Request Button to support checkout from the product detail and cart pages.
+ *
+ * Adapted from WooCommerce Stripe Gateway extension.
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use WCPay\Logger;
+
+/**
+ * WC_Payments_Payment_Request_Button_Handler class.
+ */
+class WC_Payments_Payment_Request_Button_Handler {
+	/**
+	 * Gateway settings.
+	 *
+	 * @var array
+	 */
+	public $gateway_settings;
+
+	/**
+	 * Is test mode active?
+	 *
+	 * @var bool
+	 */
+	public $testmode;
+
+	/**
+	 * This Instance.
+	 *
+	 * @var WC_Payments_Payment_Request_Button_Handler
+	 */
+	private static $instance;
+
+	/**
+	 * WC_Payments_Account instance to get information about the account
+	 *
+	 * @var WC_Payments_Account
+	 */
+	private $account;
+
+	/**
+	 * Initialize class actions.
+	 *
+	 * @param WC_Payments_Account $account Account information.
+	 */
+	public function __construct( WC_Payments_Account $account ) {
+		self::$instance = $this;
+
+		// Add default settings in case the payment request options are missing.
+		$this->gateway_settings = array_merge( self::get_default_settings(), get_option( 'woocommerce_woocommerce_payments_settings', [] ) );
```

Comment:
Thanks for the suggestion. I remember trying to do something like that, but I don't remember why I left it this way. This PR is old though.

I think it's wise to keep it like that and then add a follow-up PR. We have just a few hours until we possibly release with this feature, so it would take time for other people to test/review this and #1239 again.